assets/newspack-ui/scss/_modals.scss

```diff
@@ -127,61 +126,49 @@
 		}
 
 		&__header {
-			display: flex;
-			position: sticky;
-			align-items: flex-start;
-			justify-content: space-between;
+			align-items: center;
 			background: var( --newspack-ui-color-body-bg );
 			border-bottom: 1px solid var( --newspack-ui-color-border );
+			display: flex;
+			justify-content: space-between;
+			position: sticky;
+			top: 0;
 
 			h2 {
-				font-size: var( --newspack-ui-font-size-medium );
+				font-family: var( --newspack-ui-font-family );
+				font-size: var( --newspack-ui-font-size-s );
+				line-height: var( --newspack-ui-line-height-s );
 				margin: 0;
 			}
 		}
 
 		&__close {
-			margin: -2px -5px 0 0; // Offsets a padding in the SVG.
-			padding: 0 0 0 8px;
-			background: transparent;
```

Comment:
The modal checkout close button now inherits button styles, which I don't this is intended:

<img width="792" alt="image" src="https://github.com/Automattic/newspack-plugin/assets/7383192/3d9ec178-b72d-4fe8-a285-17fa0d2ee665">

Reply:
Oooo, thanks @adekbadek! This should now be fixed in the Blocks version of the PR, https://github.com/Automattic/newspack-blocks/pull/1713 -- I'll leave it unmerged in case anything still seems off!